examples/devices/iobox.py

```diff
@@ -0,0 +1,142 @@
+from dataclasses import dataclass
+from typing import TypedDict
+
+from softioc import builder
+
+from tickit.adapters.composed import ComposedAdapter
+from tickit.adapters.epicsadapter.adapter import EpicsAdapter
+from tickit.adapters.interpreters.command.command_interpreter import CommandInterpreter
+from tickit.adapters.interpreters.command.regex_command import RegexCommand
+from tickit.adapters.servers.tcp import TcpServer
+from tickit.core.components.component import Component, ComponentConfig
+from tickit.core.components.device_simulation import DeviceSimulation
+from tickit.core.device import Device, DeviceUpdate
+from tickit.core.typedefs import SimTime
+from tickit.utils.byte_format import ByteFormat
+
+
+class IoBoxDevice(Device):
+    """I/O box device.
+
+    The device has no inputs or outputs and interacts solely through adapters.
```

Comment:
I think the name `IoBox` may be confusing if the device doesn't do any I/O, or more accurately does one kind of I/O but not the other. Maybe `IsolatedDevice`?

Reply:
Yeah sure, happy to change it. I threw it together when I was testing things as it was useful and thought it could be useful to include. I can rename it, or I can just get rid of it in favour of the actual I/O box PR?I know you were looking at I/O box style device with the idea of a potential helper class #70

Reply:
Indeed, I seem to disagree with past me, he's always getting stuff wrong!
I have an idea: Make the IoBox in the PR capable of optionally doing either or both types of I/O, which makes it more generally useful anyway.
Have amended and updated the PR: https://github.com/dls-controls/tickit/pull/70/commits/448ce61cd700661ae19bb649f12ca9dafaec78b8

Reply:
I guess I was confusingly referring to just external I/O as I/O and considering the typical wiring of devices as something seperate.
So is the play to merge this and then follow it with that old PR?

Reply:
I've looked at the ammended PR and I'm happy for that to be the I/O box, but I would like there to be a device which doesn't have any graphed inputs or outputs, so i'll rename the one in this PR so we end up with both.